EXTENSIONS.md

```diff
@@ -0,0 +1,173 @@
+
+`ProviderConnectionStatus` is the main "Lifecycle" of your extension. The status is updated by automatically by Podman Desktop and reflected within the provider.
+
+Upon a successful start up via your `active` function within your extension, `ProviderConnectionStatus` will be reflected as 'started'. 
+
+`ProviderConnectionStatus` statuses are used in two areas, [extension-loader.ts](https://github.com/containers/podman-desktop/blob/main/packages/main/src/plugin/extension-loader.ts) and [tray-menu.ts](https://github.com/containers/podman-desktop/blob/main/packages/main/src/tray-menu.ts):
+* `extension-loader.ts`: Attempts to load the extension and sets the status accordingly (either `started`, `stopped`, `starting` or `stopping`). If an unknown error has occured, the status is set to `unknown`. `extension-loader.ts` also sends an API call to Podman Desktop to update the UI of the extension.
+* `tray-menu.ts`: If `extensionApi.tray.registerMenuItem(item);` API call has been used, a tray menu of the extension will be created. When created, Podman Desktop will use the `ProviderConnectionStatus` to indicate the status within the tray menu.
+
+#### Expanding the `extension-api` API
+
+Sometimes you'll need to add new functionality to the API in order to make an internal change within Podman Desktop. An example would be a new UI/UX component that happens within the renderer, you'd need to expand the API in order to make that change to Podman Desktop's inner-workings.
+
+In this example, we'll add a new function to simply display: "hello world" in the console.
+
+1. Add the new function to `/packages/extension-api/src/extension-api.d.ts`, under a namespace. This will make it accessible within the API when it's being called within your extension:
+
+```ts
+  export namespace foobar {
+    // ...
+    export function hello(input: string): void;
+  }
+```
+
+2. The `packages/main/src/plugin/extension-loader.ts` acts as an extension loader that defines all the actions needed by the API. Modify it to add the main functionality of `hello()` under the `foobar` namespace const:
+
+```ts
+// It's recommended you define a class that you retrieve from a separate file
+// see Podman and Kubernetes examples for implementation.
+
+// Add the class to the constructor of the extension loader
+import type { FoobarClient } from './foobar';
+
+export class ExtensionLoader {
+  // ...
+  constructor(
+    private foobarClient: FoobarClient,
+    // ...
+  ) {}
+// ..
+}
+
+// Initialize the 'foobar' client
+const foobarClient = this.foobarClient;
+
+// The "containerDesktopAPI.foobar" call is the namespace you previously defined within `extension-api.d.ts`
+const foobar: typeof containerDesktopAPI.foobar = {
+
+  // Define the function that you are implementing and call the function from the class you created.
+  hello(input: string): void => {
+    return foobarClient.hello(input);
+  },
+};
+```
```

Comment:
One step is missing here, when you return extension API instance you should add foobarClient to the returned object at the end of the createApi function.